kie-server-parent/kie-server-controller/kie-server-controller-rest/src/main/java/org/kie/server/controller/rest/RestRuntimeManagementServiceImpl.java

```diff
@@ -0,0 +1,110 @@
+            final ServerInstanceKeyList instances = super.getServerInstances(serverTemplateId);
+            String response = marshal(contentType, instances);
+            logger.debug("Returning response for get server instance with server template id '{}': {}", serverTemplateId, response);
+
+            return createCorrectVariant(response, headers, Response.Status.OK);
+        } catch (KieServerControllerIllegalArgumentException e) {
+            return createCorrectVariant(e.getMessage(), headers, Response.Status.NOT_FOUND);
+        } catch (KieServerControllerException e){
+            return createCorrectVariant(REQUEST_FAILED_TOBE_PROCESSED + e.getMessage(), headers, Response.Status.BAD_REQUEST);
+        } catch (Exception e) {
+            logger.error("Get server instances using server template id {} failed due to {}", serverTemplateId, e.getMessage(), e);
+            return createCorrectVariant("Unknown error " + e.getMessage(), headers, Response.Status.INTERNAL_SERVER_ERROR);
+        }
+
+    }
+
+    @GET
+    @Path("servers/{id}/instances/{instanceId}/containers")
+    @Produces({MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON})
+    public Response getServerInstanceContainers(@Context HttpHeaders headers, @PathParam("id") String serverTemplateId, @PathParam("instanceId") String instanceId) {
+        String contentType = getContentType(headers);
+        try {
+            logger.debug("Received get container {} for server template with id {} and instance id {}", serverTemplateId, instanceId);
+
+            ContainerList containers = super.getServerInstanceContainers(serverTemplateId, instanceId);
+            String response = marshal(contentType, containers);
+            logger.debug("Returning response for get containers for server template with id {} and instance id {}: {}", serverTemplateId, instanceId, response);
+
+            return createCorrectVariant(response, headers, Response.Status.OK);
+        } catch (KieServerControllerIllegalArgumentException e) {
+            return createCorrectVariant(e.getMessage(), headers, Response.Status.NOT_FOUND);
+        } catch (KieServerControllerException e){
+            return createCorrectVariant(REQUEST_FAILED_TOBE_PROCESSED + e.getMessage(), headers, Response.Status.BAD_REQUEST);
+        } catch (Exception e) {
+            logger.error("Get containers using server template id {} and instance id {} failed due to {}", serverTemplateId, instanceId, e.getMessage(), e);
+            return createCorrectVariant("Unknown error " + e.getMessage(), headers, Response.Status.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @GET
+    @Path("servers/{id}/containers/{containerId}")
+    @Produces({MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON})
+    public Response getServerTemplateContainers(@Context HttpHeaders headers, @PathParam("id") String serverTemplateId, @PathParam("containerId") String containerId) {
```

Comment:
Same problem here. It is not clear what this endpoint return. For me it looks like, that this endpoint should return container details. Am I right?

Reply:
Correct, it does return a list of containers. Perhaps update it to: _servers/{id}/containers/instances/{containerId}_ ?